cloudcix_primitives/geo_a_firewall_ns.py

```diff
@@ -1,18 +1,201 @@
+    3036: f'Failed to run add_inbound_drop payload (%(payload)s) on the disabled PodNet. Payload exited with status ',
+    3067: f'Failed to connect to the disabled PodNet for flush_outbound payload: ',
+    3068: f'Failed to run flush_outbound payload on the disabled PodNet. Payload exited with status ',
+    3069: f'Failed to connect to the disabled PodNet for add_outbound_set payload (%(payload)s): ',
+    3070: f'Failed to run add_outbound_set payload (%(payload)s) on the disabled PodNet. Payload exited with status ',
+    3071: f'Failed to connect to the disabled PodNet for add_outbound_drop payload (%(payload)s): ',
+    3072: f'Failed to run add_outbound_drop payload (%(payload)s) on the disabled PodNet. Payload exited with status ',
+    }
+
+    # Default config_file if it is None
+    if config_file is None:
+        config_file = '/opt/robot/config.json'
+
+
+    status, config_data, msg = load_pod_config(config_file)
+    if not status:
+      if config_data['raw'] is None:
+          return False, msg
+      else:
+          return False, msg + "\nJSON dump of raw configuration:\n" + json.dumps(config_data['raw'],
+              indent=2,
+              sort_keys=True)
+    enabled = config_data['processed']['enabled']
+    disabled = config_data['processed']['disabled']
+
+    def run_podnet(podnet_node, prefix, successful_payloads):
+        rcc = SSHCommsWrapper(comms_ssh, podnet_node, 'robot')
+        fmt = PodnetErrorFormatter(
+            config_file,
+            podnet_node,
+            podnet_node == enabled,
+            {'payload_message': 'STDOUT', 'payload_error': 'STDERR'},
+            successful_payloads
+        )
+        
+        rule_templates = {
+            'add_inbound_set': f'ip netns exec {namespace} nft add rule inet FILTER GEO_IN_ALLOW ip%(ip_version)s saddr @%(set)s accept',
+            'add_outbound_set': f'ip netns exec {namespace} nft add rule inet FILTER GEO_OUT_ALLOW ip%(ip_version)s saddr @%(set)s accept',
+        }
+
+        payloads = {
+            'flush_inbound': f'ip netns exec {namespace} nft flush chain inet FILTER GEO_IN_ALLOW',
+            'add_inbound_drop': f'ip netns exec {namespace} nft add rule inet FILTER GEO_IN_ALLOW drop',
+            'flush_outbound': f'ip netns exec {namespace} nft flush chain inet FILTER GEO_OUT_ALLOW',
+            'add_outbound_drop': f'ip netns exec {namespace} nft add rule inet FILTER GEO_OUT_ALLOW drop',
+        }
+
+        ret = rcc.run(payloads['flush_inbound'])
+        if ret["channel_code"] != CHANNEL_SUCCESS:
+            return False, fmt.channel_error(ret, f"{prefix+1}: " + messages[prefix+1]), fmt.successful_payloads
+        if ret["payload_code"] != SUCCESS_CODE:
+            return False, fmt.payload_error(ret, f"{prefix+2}: " + messages[prefix+2]), fmt.successful_payloads
+        fmt.add_successful('flush_inbound', ret)
+
+        for address_set in inbound:
+            if address_set.endswith('_V4'):
```

Comment:
One thing to note: on the Robot side, we have to make sure the sets being created from GeoIP lists all have a `_V4` or `_V6` suffix because nftables requires slightly different commands for IPv4 and IPv6 sets.

Reply:
Yes, they will. This will be managed in the IaaS API which Robot reads the data from to send to the primitives.

Reply:
Ok, good. If that's the case, this heuristic is going to work.